src/app/frontend/logs/logstoolbar/logstoolbar_controller.js

```diff
@@ -0,0 +1,149 @@
+    /** @private {!Object.<string, string, string, string>|null} */
+    this.params = this.state_.params;
+
+    // TODO(zreigz): should be LogsColorInversionParams for logsColorInversionService. Issue 153.
+    /**
+     * Service to notify logs controller if any changes on toolbar.
+     * @private {!Object}
+     */
+    this.logsColorInversionService = logsColorInversionService;
+
+    /** @export {!Array<!backendApi.ReplicaSetPodWithContainers>} */
+    this.pods = replicaSetPods.pods || [];
+
+    /**
+     * Currently chosen pod.
+     * @export {!backendApi.ReplicaSetPodWithContainers|undefined}
+     */
+    this.pod = this.findPodByName_(this.pods, this.params.podId);
+
+    /** @export {!Array<!backendApi.PodContainer>} */
+    this.containers = this.pod.podContainers || [];
+
+    /** @export {!backendApi.PodContainer|undefined} */
+    this.container = this.findContainerByName_(this.containers, this.params.container);
+
+    /**
+     * Pod creation time.
+     * @export {string}
+     */
+    this.podCreationTime = new Date(Date.parse(this.pod.startTime)).toLocaleString();
+
+    /**
+     * Namespace.
+     * @private {string}
+     */
+    this.namespace = this.params.namespace;
+
+    /**
+     * Replica Set name.
+     * @private {string}
+     */
+    this.replicaSetName = this.params.replicaSet;
+
+    /**
+     * Flag indicates state of log area color.
+     * If false: black text is placed on white area. Otherwise colors are inverted.
+     * @export {boolean}
+     * @return {boolean}
+     */
+    this.isTextColorInverted = function() {
+      return this.logsColorInversionService.getInverted();
+    };
+  }
+
+  /**
+   * Execute a code when a user changes the selected option of a pod element.
+   * @param {string} podId
+   * @return {string}
+   * @export
+   */
+  onPodChange(podId) {
+    // TODO(zreigz): state name and StateParam can not be import from logstoolbar_state. Issue 153.
+    return this.state_.transitionTo("logs", {
+      namespace: this.namespace,
+      replicaSet: this.replicaSetName,
+      podId: podId,
+      container: this.container.name,
+    });
+  }
+
+  /**
+   * Execute a code when a user changes the selected option of a container element.
+   * @param {string} container
+   * @return {string}
+   * @export
+   */
+  onContainerChange(container) {
+    // TODO(zreigz): state name and StateParam can not be import from logstoolbar_state. Issue 153.
+    return this.state_.transitionTo("logs", {
+      namespace: this.namespace,
+      replicaSet: this.replicaSetName,
+      podId: this.pod.name,
+      container: container,
+    });
+  }
+
+  /**
+   * Execute a code when a user changes logs panel color.
+   * @export
+   */
+  onTextColorChange() {
+    console.log(" before " + this.logsColorInversionService.getInverted());
+    this.logsColorInversionService.invert();
+    console.log(" after " + this.logsColorInversionService.getInverted());
+
```

Comment:
I have solution for this.
  invert() {
    this.inverted_= !!this.inverted_; //cast to boolean
    this.inverted_= !this.inverted_; // invert
  }